### PR TITLE
Added FOV interpolation; expect glitches

### PR DESCRIPTION
**Pull request by kaadmy**
This still has to be tested on Mac and Windows, and works for me on Linux.
This is my first change to Quetoo, so I was learning about the codebase a bit, too.

### CODE REVIEW

File: src/cgame/default/cg_main.c [hunk 1]
```diff
@@ -106,6 +107,8 @@ static void Cg_Init(void) {
 
 	cg_fov = cgi.Cvar("cg_fov", "110.0", CVAR_ARCHIVE, NULL);
 	cg_fov_zoom = cgi.Cvar("cg_fov_zoom", "55.0", CVAR_ARCHIVE, NULL);
+	cg_fov_interpolate = cgi.Cvar("cg_fov_interpolate", "100", CVAR_ARCHIVE,
+			"Speed to change the fov at in x100 degrees/second. 0 disables, 1 is 100 deg/sec, 0.5 is 50 deg/sec");
```
Comment (jdolan) [inline src/cgame/default/cg_main.c]: Description is wrong base on default value.

File: src/cgame/default/cg_view.c [hunk 2]
```diff
@@ -27,23 +27,53 @@
...
+			cg_fov->modified = false;
+		}
+	} else {
+		fov = target_fov;
+		cg_fov->modified = false;
+	}
+
+
+	cgi.view->fov[0] = fov / 2.0;
+
```
Comment (jdolan) [inline src/cgame/default/cg_view.c]: I'm pretty sure this could be dramatically simplified into a couple lines, and a `VectorLerp`.

File: src/cgame/default/cg_main.c (hunk 1 above)
Comment (kaadmy) [inline src/cgame/default/cg_main.c]: What do you mean by wrong base for the description?

**Comment by kaadmy** [comment]
> Use a single static uint32_t for time tracking

That's just how Cg_UpdateBob does it, I copied that.

I'll try using VectorLerp, I didn't know if I should/knew how to.

File: src/cgame/default/cg_main.c (hunk 1 above)
Comment (jdolan) [inline src/cgame/default/cg_main.c]: The description string describes values 0.0-1.0. The default value is 100. Those don't jive.

Comment (kaadmy) [inline src/cgame/default/cg_main.c]: Oh, didn't notice that.

Comment (WickedShell) [inline src/cgame/default/cg_main.c]: Not to make it more complicated, but why not just express it as the amount of time it takes to complete the zoom in? Slightly easier for users/editing I think, (especially since 1.0 = 100 deg/sec is a extremely weird mapping).

Comment (kaadmy) [inline src/cgame/default/cg_main.c]: @WickedShell I originally did that, but it can't work because it's stateless.
The zoom is just an alias to change the FOV, so it can go from anywhere to anywhere, including changing the FOV target while already in the interpolation state.
Having a fixed time to interpolate between the default FOV and zoom FOV would require the zooming to be remade as an FOV modifier/factor.

Comment (WickedShell) [inline src/cgame/default/cg_main.c]: Yeah, the alias has always seemed slightly weird to me, ideally it would just be a normal command (ideally one that just gets eaten locally by cgame without making it to the server). But that's scope creep on what you are trying to achieve :) It just has the advantage of not being dependent on the binds restoring fov.

**Comment by jdolan** [comment]
@kaadmy Thanks for the patch. I took your code and rolled with it, ironing out any glitches (I think I got them all).
